.gitmodules

```diff
@@ -4,3 +4,6 @@
 [submodule "webassets"]
 	path = webassets
 	url = https://github.com/gravitational/webassets.git
+[submodule "z3c"]
+	path = go-z3
+	url = git@github.com:xacrimon/go-z3.git
```

Comment:
why do we need a submodule? how big is this new dependency (how much will it increase the teleport binary size)? Why did we fork it? Should we fork it under the gravitational org instead of your personal github account?

Reply:
Ah, we need this submodule because building with z3 requires compiling sources contained within the submodule. For simplicity of deployment I went with the approach of static linking, this however means we must execute a build script within the submodule (containing go bindings and z3 source) in order to build the z3 static library which go links. After linking DSE it seems to increase binary size by about 10-15 MiB. We had some discussion about the build process in a predicate sync meeting and we've decided to go ahead with Z3 for now until or if we don't come up with a better solution further down the road. We'll also put Z3 features behind a build flag before we merge this to master in Q1 2023 for a preview release.

Reply:
I'm not sure @klizhentas should we move my bindings fork to gravitational? It's quite experimental but I had to fork bindings in order to add support for features which we need.

Reply:
we can use your fork, but please track your changes to be merged upstream